src/System.Buffers.Experimental/System/Buffers/BufferExtensions.cs

```diff
@@ -117,5 +118,107 @@ internal static int IndexOfStraddling(this ReadOnlySpan<byte> first, IReadOnlyMe
 
             return -1;
         }
+
+        static readonly int s_longSize = Vector<ulong>.Count;
+        static readonly int s_byteSize = Vector<byte>.Count;
+
+        public static int IndexOfVectorized(this Span<byte> buffer, byte value)
+        {
+            Debug.Assert(s_longSize == 4 || s_longSize == 2);
+
+            var byteSize = s_byteSize;
+
+            if (buffer.Length < byteSize * 2 || !Vector.IsHardwareAccelerated) return buffer.IndexOf(value);
+
+            Vector<byte> match = new Vector<byte>(value);
+            var vectors = buffer.NonPortableCast<byte, Vector<byte>>();
+            var zero = Vector<byte>.Zero;
+
+            for (int vectorIndex = 0; vectorIndex < vectors.Length; vectorIndex++)
+            {
+                var vector = vectors.GetItem(vectorIndex);
+                var result = Vector.Equals(vector, match);
+                if (result != zero)
+                {
+                    var longer = Vector.AsVectorUInt64(result);
+                    Debug.Assert(s_longSize == 4 || s_longSize == 2);
```

Comment:
Xeon Phi won't help you, it's a co-processor card you have to specifically target (usually with an intel c++ complier). I think some of the Sandy Bridge EP Xeon's have it, but I suspect it will be not a straight exposure of the registers because the AVX512 spec is a bit all over the shop.